docs/071-BDD.md

```diff
@@ -0,0 +1,550 @@
+# Behavior Driven Development
+
+Behavior Driven Development is a popular methodology of software development. It is considered to be an extension to TDD, and is much inspired by Agile practices. The main idea of choosing BDD for development process is breaking communication barer between business and technical teams. BDD encourages usage of automated testing to make all documented features of a project to be tested from the very beginning. This is why it is common to talk about BDD in the context of a test frameworks (like Codeception). However, BDD is much beyond the testing at all, and is more about managing the development process.
+
+## What is Behavior Driven Development
+
+BDD was introduced by [Dan North](https://dannorth.net/introducing-bdd/). He described it as: 
+
+>  outside-in, pull-based, multiple-stakeholder, multiple-scale, high-automation, agile methodology. It describes a cycle of interactions with well-defined outputs, resulting in the delivery of working, tested software that matters.
+
+BDD has its own evolution from the days it was born, started by replacing "test" to "should" in unit tests, and moving towards powerful tools like Cucumber and Behat, which made user stories (human readable text) to be executed as an acceptance test. 
+
+The idea of story BDD can be narrowed to:
+
+* describe features in a scenario with a formal text
+* use examples to make abstract things concrete
+* implement each step of a scenario for testing
+* write actual code implementing the feature
+
+By making every feature to be formally written (in User Story format) and test automated, we ensure that business, developers, QAs, managers are in the same boat. That everyone in a team knows what is developed, what is not, what is tested and what is not. By making tests to be a part of user story BDD allows that non-technical personnel to write (or edit) acceptance tests. BDD also encourages exploration and discussions. To formalize the requirements and write user story in a way everyone can understand them, business and developers should talk, ask questions, and play the scenarios of features to implement.
```

Comment:
... and `play` the scenarios of features to implement.

I'm not sure to understand the word "play" in this sentence.